### PR TITLE
Fix/reload picker

### LINKED ISSUE #187: [BUG][Pickers-React] Cannot navigate up
# 🐜 Bug Report

Error navigating upwards if the parent doesn't have a Workspace reference expanded

## 📦 Package / Application

pickers-react

## 📃 Version

@latest dev

## 🐜 Describe the bug

when you try to navigate upwards to a content with no Workspace ref expanded, it won't do anything. There's also some unhandled null checks.

## 👉 Steps To Reproduce

1. Go to the Admin UI
1. Navigate to a subfolder
1. Select an item
1. Select "Copy" from the context menu
1. Try to navigate upwards. You won't be able to navigate to the parent, if it has no Workspace reference.
## 😯 Current behavior

You will also get an error on the console:
```
Uncaught TypeError: Cannot read property 'Id' of null
    at setParentIdOnDoubleClick (webpack-internal:///../../packages/sn-pickers-react/dist/ListPicker/ListPicker.js:77)
    at Object.onItemDoubleClickHandler [as onDoubleClickHandler] (webpack-internal:///../../packages/sn-pickers-react/dist/ListPicker/ListPicker.js:63)
    at onDoubleClick (webpack-internal:///../../packages/sn-pickers-react/dist/ListPicker/Item.js:17)
    at HTMLUnknownElement.callCallback (webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:149)
    at Object.invokeGuardedCallbackDev (webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:199)
    at invokeGuardedCallback (webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:256)
    at invokeGuardedCallbackAndCatchFirstError (webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:270)
    at executeDispatch (webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:561)
    at executeDispatchesInOrder (webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:580)
    at executeDispatchesAndRelease (webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:680)
```

## 🤔 Expected behavior

 - I think you should be able to navigate up by default if you can load the parent content.
 - If the app developer wants to limit this functionality (e.g. allow browsing only under a specific workspace or a content with a specification) it should be set up on a prop (e.g. a ``canSetParent: (parent: GenericContent) => boolean`` callback)

## 💻 Code sample (for developers)

https://github.com/SenseNet/sn-client/blob/01708dcb33000308f3e32007012395730c61e3cc/packages/sn-pickers-react/src/ListPicker/ListPicker.tsx#L81

## 🌍 Your Environment (for developers)

| Software         | Version(s) |
| ---------------- | ---------- |
| Language (js/ts) |ts
| Operating System |win10
| NPM/Node/Yarn    |yarn
| Bundler          |webpack

## 🌍 Your environment (for apps / examples)

a frying pan

### PR DESCRIPTION
**Pull request by B3zo0**
## What I did in this pr:
- full rework of ListPicker component 💥
  - moved the picker logic to a hook - this gives way more flexibility than a component
  - reload is now triggered manually (removed repository events subscriptions)
- dms path picker refactor
  - path picker is a function component now and uses the new useListPicker hook
  - removed unnecessary codes
  - fixed some bugs related to the picker - creating a folder in path picker now shows up
  - restriction to move higher than a workspace is removed
- component docs
  - added example with hook and state reducer to picker stories
- controls react
  - avatar picker refactor to use new hook

Fixes  #187 #166

### CODE REVIEW

File: packages/sn-controls-react/src/fieldcontrols/Avatar/AvatarPicker.tsx [hunk 1]
```diff
@@ -31,94 +31,86 @@ interface AvatarPickerProps {
...
-    <ListItem button={true} selected={node.Id === this.props.selected.Id}>
-      {node.IsFolder || node.IsFolder === undefined ? (
-        <ListItemIcon>{this.props.renderIcon(this.iconName(node))}</ListItemIcon>
-      ) : (
-        <Avatar src={`${this.props.repositoryUrl}${node.Path}`} />
-      )}
-      <ListItemText primary={node.DisplayName} />
-    </ListItem>
+export function AvatarPicker(props: AvatarPickerProps) {
+  // TODO? handle error and loading?
```
Comment (herflis) [inline packages/sn-controls-react/src/fieldcontrols/Avatar/AvatarPicker.tsx]: :shipit:

File: packages/sn-pickers-react/src/ListPicker/useListPicker.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  return parent && parent.Id === node.Id
+    ? { parentId: parent.ParentId, path: parent.Path }
+    : { parentId: node.ParentId, path: node.Path }
+}
+
+/**
+ * useListPicker let you select and navigate in the reposiotry with built in defaults
+ */
+export const useListPicker = <T extends GenericContent = GenericContent>(
+  repository: Repository,
```
[the remark this answers is not in the document]
Comment (B3zo0) [reply packages/sn-pickers-react/src/ListPicker/useListPicker.ts]: yep ✔

File: packages/sn-controls-react/src/fieldcontrols/Avatar/AvatarPicker.tsx (hunk 1 above)
Comment (B3zo0) [reply packages/sn-controls-react/src/fieldcontrols/Avatar/AvatarPicker.tsx]: added a spinner and error handling

**Comment by kultsar** [comment]
- [ ] Disable Copy/Move button in dialogue when user selects parent folder.

**Pull request merged**